pkg/controller/controlplane/valuesprovider.go

```diff
@@ -750,12 +751,139 @@ func getCSIControllerChartValues(
 			},
 		},
 	}
+
+	if csi != nil {
+		if csi.CSIAttacher != nil {
+			values["csiAttacher"] = getCSIAttacherArgs(csi.CSIAttacher)
+		}
+		if csi.CSIResizer != nil {
+			values["csiResizer"] = getCSIResizerArgs(csi.CSIResizer)
+		}
+		if csi.CSIDriverCinder != nil {
+			values["csiDriverCinder"] = getCSIDriverCinderArgs(csi.CSIDriverCinder)
+		}
+		if csi.CSISnapshotController != nil {
+			values["csiSnapshotController"] = getCSISnapshotControllerArgs(csi.CSISnapshotController)
+		}
+		if csi.CSILivenessProbe != nil {
+			values["csiLivenessProbe"] = getCSILivenessProbeArgs(csi.CSILivenessProbe)
+		}
+		if csi.CSIProvisioner != nil {
+			values["csiProvisioner"] = getCSIProvisionerArgs(csi.CSIProvisioner)
+		}
+		if csi.CSISnapshotter != nil {
+			values["csiSnapshotter"] = getCSISnapshotterArgs(csi.CSISnapshotter)
+		}
+	}
+
 	if userAgentHeaders != nil {
 		values["userAgentHeaders"] = userAgentHeaders
 	}
 	return values, nil
 }
 
+func getCSISnapshotterArgs(snapshotter *config.CSISnapshotter) map[string]interface{} {
+	csiSnapshotter := make(map[string]interface{})
+	csiSnapshotterArgs := make(map[string]interface{})
+	if snapshotter.Verbose != nil {
+		csiSnapshotterArgs["verbosity"] = snapshotter.Verbose
+	}
+	if snapshotter.Timeout != nil {
+		csiSnapshotterArgs["timeout"] = snapshotter.Timeout
+	}
+	csiSnapshotter["args"] = csiSnapshotterArgs
+	return csiSnapshotter
+}
+
+func getCSIProvisionerArgs(provisioner *config.CSIProvisioner) map[string]interface{} {
+	csiProvisioner := make(map[string]interface{})
+	csiProvisionerArgs := make(map[string]interface{})
+	if provisioner.Verbose != nil {
+		csiProvisionerArgs["verbosity"] = provisioner.Verbose
+	}
+	if provisioner.Timeout != nil {
+		csiProvisionerArgs["timeout"] = provisioner.Timeout
+	}
```

Comment:
Once we embed these fields in the api we can move this in a dedicated function and just call it in each `getCSI...Args()` func.